services/QuillLMS/app/services/vitally_integration/serialize_vitally_sales_organization.rb

```diff
@@ -165,6 +193,31 @@ def activities_completed_all_time
         .where('activity_sessions.state = ?', 'finished')
     end
 
+    def activities_assigned_query(district)
+      ClassroomUnit.joins("JOIN unit_activities ON classroom_units.unit_id=unit_activities.unit_id")
+      .joins("JOIN activities ON activities.id = unit_activities.activity_id")
+      .joins("JOIN classrooms ON classrooms.id = classroom_units.classroom_id")
+      .joins("JOIN classrooms_teachers ON classrooms.id=classrooms_teachers.classroom_id")
+      .joins("JOIN schools_users ON schools_users.user_id = classrooms_teachers.user_id")
+      .joins("JOIN schools ON schools_users.school_id=schools.id")
+      .joins("JOIN districts ON schools.district_id = districts.id")
+      .where("districts.id = ?", district.id)
+      .select("assigned_student_ids, activities.id, unit_activities.created_at")
```

Comment:
Same as a previous comment: this is probably re-writeable using defined model relationships.